lib/Data/S3Storage.php

```diff
@@ -82,31 +82,33 @@ class S3Storage extends AbstractData
      */
     public function __construct(array $options)
     {
-        $this->_options['credentials'] = array();
-
-        if (is_array($options) && array_key_exists('region', $options)) {
-            $this->_options['region'] = $options['region'];
-        }
-        if (is_array($options) && array_key_exists('version', $options)) {
-            $this->_options['version'] = $options['version'];
-        }
-        if (is_array($options) && array_key_exists('endpoint', $options)) {
-            $this->_options['endpoint'] = $options['endpoint'];
-        }
-        if (is_array($options) && array_key_exists('accesskey', $options)) {
-            $this->_options['credentials']['key'] = $options['accesskey'];
-        }
-        if (is_array($options) && array_key_exists('secretkey', $options)) {
-            $this->_options['credentials']['secret'] = $options['secretkey'];
-        }
-        if (is_array($options) && array_key_exists('use_path_style_endpoint', $options)) {
-            $this->_options['use_path_style_endpoint'] = filter_var($options['use_path_style_endpoint'], FILTER_VALIDATE_BOOLEAN);
-        }
-        if (is_array($options) && array_key_exists('bucket', $options)) {
-            $this->_bucket = $options['bucket'];
-        }
-        if (is_array($options) && array_key_exists('prefix', $options)) {
-            $this->_prefix = $options['prefix'];
+        if (is_array($options)) {
```

Comment:
Moved up this conditional, as it was being repeated in every single check below. If this was an intentional design choice, I'm happy to move it back.

Reply:
Good cleanuü, next time I would even argue to folllow the [early return](https://dev.to/i3uckwheat/early-return-pattern-558i) pattern and not intend that stuff so much.